src/main/java/com/google/plantasy/utils/QuizTimingPropertiesUtils.java

```diff
@@ -71,14 +71,22 @@ public static Long getQuizTimestampProperty(String entity, String id_of_entity,
         return null;
     }
 
-    //This function checks if the user has taken the quiz yet by comparing their timestamp with the quiz's timestamp
-    public static boolean userTookQuiz(Long usersQuizTime, Long currentQuizTime) {
+    //This function checks if the user has taken the quiz yet by comparing their timestamp with today's timestamp
+    public static boolean userTookQuiz(Long usersQuizTime) {
         SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
         sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
-        String users_quiz_time = sdf.format(usersQuizTime);
-        String current_quiz_time = sdf.format(currentQuizTime);
+        String users_quiz_time;
+        String today_time;
 
-        return(users_quiz_time.compareTo(current_quiz_time) > 0);
+        try {
+            users_quiz_time = sdf.format(usersQuizTime);
+            today_time = sdf.format(new Date());
+        } catch(IllegalArgumentException e ) {
+            log.log(Level.SEVERE, "Null result for parameter");
+            return false;
+        }  
+
+        return (users_quiz_time.equals(today_time));
```

Comment:
doing equals works here but the logic is a bit confusing. if i understand correctly, this is allowed because the user quiz time will never be greater than today's date (because the user can't time travel) so just checking to see if it is equal is sufficient. If this is the case, maybe add a small comment stating as such to avoid confusion. 

Also, you can ditch the extra pair on () on this line (line 89)

Reply:
Added some comments, let me know if that makes it more clear.